services/transactionService.ts

```diff
@@ -745,31 +730,40 @@ export async function fetchAllPaymentsByUserIdWithPagination (
 
 export async function fetchAllPaymentsByUserId (
   userId: string,
-  networkIds?: number[]
+  networkIds?: number[],
+  buttonIds?: string[]
 ): Promise<TransactionsWithPaybuttonsAndPrices[]> {
-  const transactions = await prisma.transaction.findMany({
-    where: {
-      address: {
-        userProfiles: {
-          some: {
-            userId
-          }
-        },
-        networkId: {
-          in: networkIds ?? Object.values(NETWORK_IDS)
-        }
+  const whereClause: any = {
+    address: {
+      userProfiles: {
+        some: { userId }
       },
-      amount: {
-        gt: 0
+      networkId: {
+        in: networkIds ?? Object.values(NETWORK_IDS)
       }
     },
+    amount: {
+      gt: 0
+    }
+  }
+
+  if (Array.isArray(buttonIds) && buttonIds.length > 0) {
```

Comment:
In the function above we had similar code but with `if ((buttonIds != null)`, which was basically checking if `buttonIds !== undefined`. This `Array.isArray(buttonIds)` achieves the same result but with in a different way, which makes what it is doing a little less obvious and harder to read.